src/main/java/org/folio/linked/data/configuration/audit/LinkedDataAuditEntityListener.java

```diff
@@ -0,0 +1,28 @@
+package org.folio.linked.data.configuration.audit;
+
+import static java.util.Objects.isNull;
+import static java.util.Optional.ofNullable;
+
+import jakarta.persistence.PrePersist;
+import lombok.AllArgsConstructor;
+import org.folio.linked.data.model.entity.Resource;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.data.auditing.AuditingHandler;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+import org.springframework.stereotype.Component;
+
+@Component
+@AllArgsConstructor
+public class LinkedDataAuditEntityListener extends AuditingEntityListener {
+
+  private ObjectFactory<AuditingHandler> handler;
+
+  @Override
+  @PrePersist
+  public void touchForCreate(Object target) {
+    if (target instanceof Resource resource && isNull(resource.getCreatedBy())) {
+      ofNullable(handler.getObject())
```

Comment:
`handler` is nullable. So, `handler.getObject()` may through NullPointerException. The `ofNullable(handler.getObject())` will not help.

May be it be better to do like this?
```
    if (target instanceof Resource resource && isNull(resource.getCreatedBy())) {
      super.touchForCreate(target);
    }
```

Reply:
have updated the code